jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/handler/ResourceHandlerTest.java

```diff
@@ -750,8 +749,35 @@ public void testBigger() throws Exception
     }
 
     @Test
-    @Disabled
-    public void testBrotli() throws Exception
+    public void testBrotliInitialCompressed() throws Exception
+    {
+        Files.writeString(docRoot.resolve("data0.txt"), "Hello Text 0", UTF_8);
+        Files.writeString(docRoot.resolve("data0.txt.br"), "fake brotli", UTF_8);
+
+        _rootResourceHandler.setDirAllowed(false);
+        _rootResourceHandler.setRedirectWelcome(false);
+        _rootResourceHandler.setPrecompressedFormats(CompressedContentFormat.BR);
+        _rootResourceHandler.setEtags(true);
+
+        String rawResponse = _local.getResponse("""
+            GET /context/data0.txt HTTP/1.1\r
+            Host: localhost:8080\r
+            Connection: close\r
+            Accept-Encoding: gzip;q=0.9,br\r
+            \r
+            """);
+        HttpTester.Response response = HttpTester.parseResponse(rawResponse);
+        assertThat(response.toString(), response.getStatus(), is(HttpStatus.OK_200));
+        assertThat(response, containsHeaderValue(HttpHeader.CONTENT_LENGTH, "11"));
+        assertThat(response, containsHeaderValue(HttpHeader.CONTENT_TYPE, "text/plain"));
+        assertThat(response, containsHeaderValue(HttpHeader.VARY, "Accept-Encoding"));
+        assertThat(response, containsHeaderValue(HttpHeader.CONTENT_ENCODING, "br"));
+        String body = response.getContent();
+        assertThat(body, containsString("fake br"));
```

Comment:
Why only `fake br`?

Reply:
The contents don't actually have to be compressed to satisfy the test.
The ResourceService doesn't compress it anyway, it just serves the content as-is from the Resource Base.

Reply:
But it confuses the reader the fact that you test for `fake br` only, while at the beginning of the test the whole content is `fake brotli`. 
I would test for the whole content, since it's so small, to avoid such scratching head moment from other readers of this test ("why only `fake br`"?).